merge_command.go

```diff
@@ -52,19 +56,80 @@ func mergeWithRetry(nrOfRetries int, issueComment IssueComment, issues Issues, p
 		log.Printf("PR #%d has pending and/or failed statuses. Not merging.\n", issueComment.IssueNumber)
 		return SuccessResponse{}
 	}
-	err := merge(issueComment.Repository, issueComment.IssueNumber, pullRequests)
-	if err == ErrOutdatedMergeRef && nrOfRetries > 0 {
-		return mergeWithRetry(nrOfRetries-1, issueComment, issues, pullRequests, repositories, gitRepos)
+	if errResp = mergeReadyPR(issueComment.Issue(), issues, pullRequests); errResp != nil {
+		return errResp
+	}
+	return SuccessResponse{fmt.Sprintf("Successfully merged PR %s", issueComment.Issue().FullName())}
+}
+
+func mergeReadyPR(issue Issue, issues Issues, pullRequests PullRequests) *ErrorResponse {
+	err := merge(issue.Repository, issue.Number, pullRequests)
+	if err == ErrMergeConflict {
+		return handleMergeConflict(issue, issues)
 	} else if err != nil {
-		message := fmt.Sprintf("Failed to merge PR #%d", issueComment.IssueNumber)
-		return ErrorResponse{err, http.StatusBadGateway, message}
+		message := fmt.Sprintf("Failed to merge PR %s", issue.FullName())
+		return &ErrorResponse{err, http.StatusBadGateway, message}
 	}
-	log.Printf("PR #%d successfully merged. Removing the '%s' label.\n", issueComment.IssueNumber, MergingLabel)
-	errResp = removeLabel(issueComment.Repository, issueComment.IssueNumber, MergingLabel, issues)
+	log.Printf(
+		"PR %s successfully merged. Removing the '%s' label.\n",
+		issue.FullName(),
+		MergingLabel,
+	)
+	errResp := removeLabel(issue.Repository, issue.Number, MergingLabel, issues)
 	if errResp != nil {
 		return errResp
 	}
-	return SuccessResponse{}
+	return nil
+}
+
+func checkForMergeOptions(statusEvent StatusEvent, search Search, issues Issues,
```

Comment:
But it's only an indication that there might be some ready. An additional search must be done to see if any PR is actually ready for merging.